call-notes/Epoch Rewards.md

```diff
@@ -0,0 +1,171 @@
+
+
+**Attendee 1** :  Okay. And I do have another question, but I don't want to monopolize this, so I'll wait to see if other people have a question first.
+
+**Jacob Creech**: All right, next question.
+
+I think it was Jeff, so go ahead, Jeffy, you should be able to unmute to ask your question. 
+
+**Jeff** : Oh, sorry, my mistake. 
+
+**Jacob Creech**: Okay, is there any other questions about this, or otherwise we'll go back to you, Zans. All right, Ashwin, go ahead.
+
+**Aswin**: Hey just wanted to ask, if there's a cluster we started in the first thousand blocks, is there any change to change that procedure because the snapshot won't have the result in it? 
+
+**Haoran** : Yeah, we have a talk about this, so in that case, when the snapshot is taken during the reward computation phase, the snapshot has to wait, it has to wait until the result is available and store the result there.
+
+**Aswin**: Got it. 
+
+**Jacob Creech**: Okay, Zans, you're welcome to ask your question. 
+
+**Zantetsu**: Okay, so is it the case that validators will only validate a block if the leader producing that block has the appropriate amount of either computation work or rewards pair of work in that block? Or is it the case that a validator can just sort of decide not to do it on their block and be like, "I've modified the code to not do this because I don't want to burn the CPU on this and it doesn't give me any rewards, so I'm just not going to do it?" Or is that even a possibility? Or is it the case that the cluster won't even accept a block unless it includes these details because they're expected for... I mean, they're sort of built into the protocol that this has to happen in this order. 
+
+**Haoran**: Yes, because the order is deterministic  for the current epoch, but there will be some randomness across different epochs. So if the leader, for example, doesn't include the expected set of rewards in the block, I think the hashes will mismatch and his block will be rejected. 
+
+**Zantetsu**: Okay, so this part of the block does go into the hash so that validators have to agree on it to vote on it. So I guess that answers my question and we just kind of accept that the first leader kind of has to do his work without any compensation. That's just kind of built into the protocol. There's no need to add any compensation for doing this work. It's not expected to impact the transactions that could be fit into a block. Like, it doesn't take up extra block space or extra compute time that could prevent a validator from including as many transactions as it would otherwise.
+
+
+**Attendee 4**:  It's the same problem as we have right now with only one, whatever that first leader or like whatever that first block is, kind of gets screwed on including transactions because everyone is computing this. So you'd have this, it's just spreading out the computation to make it more feasible. 

```

Comment:
```suggestion
**Anatoly Yakovenko**:  It's the same problem as we have right now with only one, whatever that first leader or like whatever that first block is, kind of gets screwed on including transactions because everyone is computing this. So you'd have this, it's just spreading out the computation to make it more feasible. 

```